governance/src/components/Profile/ProfileActivity.tsx

```diff
@@ -0,0 +1,316 @@
+      switch (activityFilter) {
+        case "all activity":
+          return true;
+        case "allocate voting power":
+          return activity.type === "allocateVote";
+        default:
+          return activity.type === activityFilter;
+      }
+    });
+
+    filteredActivities.sort((a, b) => {
+      switch (sortBy) {
+        case "latest first":
+          return a.timestamp < b.timestamp ? 1 : -1;
+        case "oldest first":
+        default:
+          return a.timestamp > b.timestamp ? 1 : -1;
+      }
+    });
+
+    return filteredActivities;
+  }, [activities, activityFilter, sortBy]);
+
+  const getActivityLogoColor = useCallback((activity: ActivityType) => {
+    switch (activity.type) {
+      case "allocateVote":
+        return `${getVaultColor(activity.vault!)}1F`;
+      case "stake":
+        return `${colors.red}1F`;
+      case "vote":
+        return activity.vote === "for"
+          ? `${colors.green}1F`
+          : `${colors.red}1F`;
+      default:
+        return undefined;
+    }
+  }, []);
+
+  const renderActivityLogo = useCallback((activity: ActivityType) => {
+    switch (activity.type) {
+      case "allocateVote":
+        const asset = getDisplayAssets(activity.vault!);
+        const Logo = getAssetLogo(asset);
+        switch (asset) {
+          case "WETH":
+            return <Logo height="70%" />;
+          default:
+            return <Logo />;
+        }
+      case "vote":
+        switch (activity.vote) {
+          case "for":
+            return <CheckIcon color={colors.green} />;
+          default:
+            return <CloseIcon color={colors.red} />;
+        }
+      case "stake":
+        return <StakeIcon color={colors.red} />;
+      case "unstake":
+        return <UnstakeIcon />;
+      case "delegate":
+        return <DelegateIcon width="20px" />;
+    }
+  }, []);
+
+  const getActivityTitle = useCallback(
+    (activity: ActivityType) => {
+      switch (activity.type) {
+        case "allocateVote":
+          return "ALLOCATED VOTING POWER";
+        case "vote":
+          return `VOTED ${
+            activity.vote === "for" ? "FOR" : "AGAINST"
+          } PROPOSAL ${activity.proposal}`;
+        case "stake":
+        case "unstake":
+          return `${activity.type}D`;
+        case "delegate":
+          return (
+            <>
+              DELEGATED VOTES <span style={{ color: colors.text }}>TO</span>{" "}
+              {activity.address === account
+                ? "SELF"
+                : truncateAddress(activity.address!)}
+            </>
+          );
+        default:
+          return activity.type;
+      }
+    },
+    [account]
+  );
+
+  const renderActivityData = useCallback((activity: ActivityType) => {
+    switch (activity.type) {
+      case "stake":
+      case "unstake":
+        return (
+          <Title>
+            {formatAmount(parseFloat(formatUnits(activity.amount!, 18)))} RBN
```

Comment:
Could we refactor these function calls into `formatBigNumber` or something?

Reply:
Actually `formatBigNumber` and `formatAmount` are quite different in their utility in the respect where `formatAmount` turns the number into such as `138K` or `28M` whereas `formatBigNumber` would have to turn them into `138,398` and `28,284,283`  respectively. The reason I am using `formatUnits` here before passing into `formatAmount` is that `formatBigNumber` returns back number in a string that had been "tainted" with the dot `,`, and `formatAmount` require them in number format.

Reply:
But I guess we can have another utility that takes in BigNumber and return them in the `K` and `M` format